Add subscription central info to Managing RH subscriptions (#3702)
theforeman/foreman-documentation: guides/common/modules/con_managing-red-hat-subscriptions.adoc

[id="Managing_Red_Hat_Subscriptions_{context}"]
= Managing Red Hat subscriptions

{ProjectName} can import content from the {Team} Content Delivery Network (CDN).
{Project} requires a {Team} subscription manifest to find, access, and download content from the corresponding repositories.
You must have a {Team} subscription manifest containing a subscription allocation for each organization on {ProjectServer}.

> All subscription information is available in your {Team} Customer Portal account.

Reviewer: Can we stop referencing the deprecated Customer Portal? We should direct users to console.redhat.com instead.

> All subscription information is available in your {Team} Customer Portal account.

Reviewer 2: This text is available in upstream docs as well. Please, don't replace `Red{nbsp}Hat` with the `{Team}` attribute, it has to be hard coded, otherwise it will result in nonsensical concepts, such as "Foreman community CDN", which doesn't exist.

.Subscription allocations and organizations
You can manage more than one organization if you have more than one subscription allocation.
{Project} requires a single allocation for each organization configured in {ProjectServer}.
The advantage of this is that each organization maintains separate subscriptions so that you can support multiple organizations, each with their own {Team} accounts.

.Future-dated subscriptions
You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Subscriptions service capabilities
* Allows customers to view subscription usage.
* When configured, it displays usage data from both connected and disconnected environments.
* Tracks {Team} product usage and capacity at the fleet or account level.
* Generates a daily snapshot of usage data across all tracked systems.
* Displays data in a filterable dashboard at cloud.redhat.com.
* Collects historical data for analytics, purchasing decisions, capacity planning, and risk mitigation.

> * Supports the transition to Simple Content Access (SCA), which enforces subscriptions across the organization rather than on individual systems.

Reviewer: This just jumps right into a bulleted list without explaining what the Subscriptions service is or why I am reading this. Is the Subscriptions service a capability of Satellite? (no it isn't, but I wouldn't know that from reading this.)

.Configuring your system to use the Subscriptions service

Connected (Non-{Project}) users::

> As a connected, non-{Project} user, no further action needs to be taken to use the subscriptions service. 

Reviewer: Why would we include information for non-Satellite users in the Satellite documentation?

To track your subscription usage, log in to the Hybrid Cloud Console at https://console.redhat.com/subscriptions/usage/rhel. 
Registering a system through the guidance in https://docs.redhat.com/en/documentation/subscription_central/1-latest/html-single/getting_started_with_rhel_system_registration/index[Getting Started with RHEL System Registration] is sufficient to ensure the system is properly counted. 
It is recommended to set System Purpose attributes to define SLA and USAGE.
This improves filtering and enhances reporting in the Subscriptions service.
For more information, see {RHELDocsBaseURL}9/html/automatically_installing_rhel/proc_configuring-system-purpose-using-the-subscription-manager-command-line-tool_rhel-installer[Configuring System Purpose using the subscription-manager command-line tool].

ifndef::satellite[]
Connected {Project} Users::

> Connected {Project} Users should enable the `foreman_rh_cloud` plugin if it has not been enabled already (https://docs.redhat.com/en/documentation/subscription_central/1-latest/html/getting_started_with_the_subscriptions_service/proc-installing-satellite-inventory-upload-plugin_assembly-setting-up-subscriptionwatch-ctxt). The foreman_rh_cloud plugin shares inventory with the Hybrid Cloud Console enabling the usage of the subscriptions service to track usage information.

Reviewer: I'm pretty sure foreman_rh_cloud has been enabled by default in Satellite for quite some time. Should double-check this and, if it's the case, we would not need to tell users to enable it. Rather, one just needs to configure it to one's preferences.

The plugin can be configured to omit data that is not pertinent for subscription purposes, such as host names and IP addresses.
endif::[]

Disconnected users::
Users who cannot or do not want to connect directly can export a local summary of their product usage and import it into a CMDB, spreadsheet, or other tracking tools.

Disconnected (No {Project})::
Disconnected users can generate a local report using tools such as {Project}. 
Disconnected {Project} users can use `foreman_rh_cloud` reports or the {Project} host installed products report.
Users without {Project} can query standalone RHEL systems with {Team} Discovery.
For more information about {Team} Discovery, see link:https://docs.redhat.com/en/documentation/subscription_central/1-latest/html-single/installing_and_configuring_discovery/index#con-what-is-prod-common_assembly-about-common-ctxt[What is Discovery?].
Alternatively, you can leverage any tool, whether provided by {Team} or another source, to track your subscription usage.
Some users prefer to review usage reports before sending them to the Hybrid Cloud Console. 

> Discovery allows users to upload reports manually as needed.

Reviewer: Again, let's leave out non-Satellite information from Satellite docs

Disconnected ({Project})::
{Project} users can export usage data in one of two ways:

** Through the `foreman_rh_cloud` plugin, which generates its report locally and allows you the ability to download it. 
This report is in JSON format and is most suitable for automated scripts or machine ingestion. 
This report can also be generated through the `foreman-rake` command.

> For more information, see link:https://access.redhat.com/solutions/1752433[Satellite 6 : Running foreman-rake console with the possibility to rollback any changes].

Reviewer: The rake console is not what you're looking for here. (In other words, this link is incorrect.)
To generate the report I think you'd run
```
# foreman-rake rh_cloud_inventory:report:generate
```

Author: Does the user need to run this on root?

.Future-dated subscriptions
You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Subscriptions service capabilities
The Subscriptions service is a service that helps organizations track and manage their Red{nbsp}Hat subscription usage across connected and disconnected environments. 
* Allows customers to view subscription usage.
* When configured, it displays usage data from both connected and disconnected environments.
* Tracks Red{nbsp}Hat product usage and capacity at the fleet or account level.
* Generates a daily snapshot of usage data across all tracked systems.
* Displays data in a filterable dashboard at cloud.redhat.com.
* Collects historical data for analytics, purchasing decisions, capacity planning, and risk mitigation.
* Supports the transition to Simple Content Access (SCA), which enforces subscriptions across the organization rather than on individual systems.

.Configuring your system to use the Subscriptions service

> ifndef::satellite[]

Reviewer: I didn't realize this `ifndef::satellite` was here. Upstream users will not have the foreman_rh_cloud plugin enabled by default, so we need to include the foreman-installer command to enable it, as a prerequisite.

ifndef::satellite[]
Connected {Project} Users::
Connected {Project} Users should configure the `foreman_rh_cloud` plugin to share inventory information with the Hybrid Cloud Console.

> The plugin enables the usage of the subscriptions service to track usage information across connected systems.

Reviewer: How?

Should reference the automatic inventory upload setting.

Author: Should it say `Ensure the **Automatic Inventory Upload** setting is enabled`?

Reviewer: Yes :+1:

.Future-dated subscriptions
You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Subscriptions service capabilities
The Subscriptions service is a service that helps organizations track and manage their Red{nbsp}Hat subscription usage across connected and disconnected environments. 
* Allows customers to view subscription usage.
* When configured, it displays usage data from both connected and disconnected environments.
* Tracks Red{nbsp}Hat product usage and capacity at the fleet or account level.
* Generates a daily snapshot of usage data across all tracked systems.
* Displays data in a filterable dashboard at cloud.redhat.com.
* Collects historical data for analytics, purchasing decisions, capacity planning, and risk mitigation.
* Supports the transition to Simple Content Access (SCA), which enforces subscriptions across the organization rather than on individual systems.

.Configuring your system to use the Subscriptions service

> 

Reviewer 3: ```suggestion
```

ifndef::satellite[]
Connected {Project} Users::
Connected {Project} Users should configure the `foreman_rh_cloud` plugin to share inventory information with the Hybrid Cloud Console.
Ensure the *Automatic Inventory Upload* setting is enabled.
The plugin enables the usage of the subscriptions service to track usage information across connected systems.

The plugin can be configured to omit data that is not pertinent for subscription purposes, such as host names and IP addresses.
endif::[]

Disconnected users::
Users who cannot or do not want to connect directly can export a local summary of their product usage and import it into a CMDB, spreadsheet, or other tracking tools.

Disconnected ({Project})::
{Project} users can export usage data in one of two ways:

> ** Through the `foreman_rh_cloud` plugin, which generates its report locally and allows you the ability to download it. 

Reviewer 3: ```suggestion
** Through the `foreman_rh_cloud` plugin, which generates its report locally and allows you to download it. 
```

suggestion (non-blocking): reword it slightly.

This report is in JSON format and is most suitable for automated scripts or machine ingestion. 

> This report can also be generated through the `foreman-rake` command.

Reviewer 3: I am unsure about the exact wording, esp. "by using CLI", but I am trying to use active language:

```suggestion
You can generade this report by using CLI:
```

alternative:
```suggestion
Alternatively, you can generade this report by using CLI:
```

For the second suggestion, you might want to extend line 47 with "... from the {ProjectWebUI}".

{Project} requires a Red{nbsp}Hat subscription manifest to find, access, and download content from the corresponding repositories.
You must have a Red{nbsp}Hat subscription manifest containing a subscription allocation for each organization on {ProjectServer}.
All subscription information is available on `console.redhat.com`.

Use this chapter to import a Red{nbsp}Hat subscription manifest and manage the manifest within the {ProjectWebUI}.

.Subscription allocations and organizations
You can manage more than one organization if you have more than one subscription allocation.
{Project} requires a single allocation for each organization configured in {ProjectServer}.
The advantage of this is that each organization maintains separate subscriptions so that you can support multiple organizations, each with their own Red{nbsp}Hat accounts.

.Future-dated subscriptions
You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Subscriptions service capabilities

> The Subscriptions service is a service that helps organizations track and manage their Red{nbsp}Hat subscription usage across connected and disconnected environments. 

Reviewer 2: ```suggestion
The Subscriptions service is a service on the {RHCloud} that helps organizations track and manage their Red{nbsp}Hat subscription usage across connected and disconnected environments. 
```
I'd be more specific about where that service lives :)

* Allows customers to view subscription usage.
* When configured, it displays usage data from both connected and disconnected environments.

> * Tracks Red{nbsp}Hat product usage and capacity at the fleet or account level.

Reviewer 2: I'm unsure about what "fleet" means in this context. I'm aware that in the Cloud Console, there's a RH account and an organization to which accounts can belong. Does "fleet level" mean "organization level"?

Author: Yes I will change to **organization** level.

* Generates a daily snapshot of usage data across all tracked systems.

> * Displays data in a filterable dashboard at cloud.redhat.com.

Reviewer 2: ```suggestion
* Displays data in a filterable dashboard at the {RHCloud}.
```

* Collects historical data for analytics, purchasing decisions, capacity planning, and risk mitigation.
* Supports the transition to Simple Content Access (SCA), which enforces subscriptions across the organization rather than on individual systems.

.Configuring your system to use the Subscriptions service
ifndef::satellite[]
.Prerequisites
* Enable the `foreman_rh_cloud` plugin:
[subs="nowrap" subs="+quotes,verbatim,attributes"]
----
# foreman-installer --enable-foreman-plugin-rh-cloud

> ----

Reviewer 2: Isn't the plugin enabled by default in Satellite? Perhaps this is only necessary in upstream? Might need an `ifdef` condition.

Author: It has the `ifndef` condition.

Connected {Project} Users::
Connected {Project} Users should configure the `foreman_rh_cloud` plugin to share inventory information with the Hybrid Cloud Console.
Ensure the *Automatic Inventory Upload* setting is enabled.
The plugin enables the usage of the subscriptions service to track usage information across connected systems.

The plugin can be configured to omit data that is not pertinent for subscription purposes, such as host names and IP addresses.
endif::[]

Disconnected users::
Users who cannot or do not want to connect directly can export a local summary of their product usage and import it into a CMDB, spreadsheet, or other tracking tools.

Disconnected ({Project})::

> {Project} users can export usage data in one of two ways:

Reviewer 2: These parts aren't user-focused enough. Instead of speaking about "users", speak to the user directly.

The advantage of this is that each organization maintains separate subscriptions so that you can support multiple organizations, each with their own Red{nbsp}Hat accounts.

.Future-dated subscriptions

You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Prerequisites
* Ensure you have a Red{nbsp}Hat subscription manifest.
ifdef::orcharhino[]
You can create and download your manifest in the https://access.redhat.com/management/subscription_allocations[Red Hat Customer Portal].
endif::[]
ifndef::orcharhino[]
** If your {Project} is connected, use the {RHCloud} to create the manifest.
ifdef::satellite[]
For more information, see {RHDocsBaseURL}subscription_central/1-latest/html-single/creating_and_managing_manifests_for_a_connected_satellite_server/index[Creating and managing manifests for a connected {ProjectServer}] in _Subscription Central_.
endif::[]
** If your {Project} is disconnected, use the Red Hat Customer Portal to create the manifest.
ifdef::satellite[]

> For more information, see {RHDocsBaseURL}subscription_central/1-latest/html/getting_started_with_rhel_system_registration/adv-reg-rhel-using-rhsm_#using_manifests_con[Using manifests for a disconnected Satellite Server] in _Subscription Central_.

Reviewer 2: The [SAT-29714](https://issues.redhat.com/browse/SAT-29714) Jira issue requested these links to be replaced by procedures. Where are those procedures?

If you plan to add them in another PR, can we keep these links for now?

Author: Yes

> .Subscriptions service capabilities

Reviewer 2: ```suggestion
.Subscriptions service
```
Looks like fluff now. Otherwise, good stuff!

The advantage of this is that each organization maintains separate subscriptions so that you can support multiple organizations, each with their own Red{nbsp}Hat accounts.

.Future-dated subscriptions
You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Subscriptions service capabilities
The Subscriptions service is a service on the {RHCloud} that helps organizations track and manage their Red{nbsp}Hat subscription usage across connected and disconnected environments. 

* Allows customers to view subscription usage.
* When configured, it displays usage data from both connected and disconnected environments.
* Tracks Red{nbsp}Hat product usage and capacity at the organization or account level.
* Generates a daily snapshot of usage data across all tracked systems.
* Displays data in a filterable dashboard at the {RHCloud}.
* Collects historical data for analytics, purchasing decisions, capacity planning, and risk mitigation.

> * Supports the transition to Simple Content Access (SCA), which enforces subscriptions across the organization rather than on individual systems.

Reviewer 2: I would either

* Move the capabilities into the procedure module, or
* Link to the procedure module from here

Please, consider which option is more suitable. I'm not sure. Perhaps we should say something about the Subscriptions service in this concept module, but it feels like the details should stick together with the procedure.

Author: I'm rewriting the capabilities as an intro without a list in the concept and then linking to the procedure.

.Configuring your system to use the Subscriptions service
ifndef::satellite[]
.Prerequisites
* Enable the `foreman_rh_cloud` plugin:
[subs="nowrap" subs="+quotes,verbatim,attributes"]
----

> # foreman-installer --enable-foreman-plugin-rh-cloud

Reviewer 4: This is only relevant for upstream, downstream we ship it enabled.

Author: This section is already conditionalized with ifndef::satellite[], so it should only appear in upstream builds.

Configure the `foreman_rh_cloud` plugin to share inventory information with the Hybrid Cloud Console.
Ensure that the *Automatic Inventory Upload* setting is enabled.
The plugin enables the subscriptions service to track usage information across connected systems.

You can configure the plugin to omit data that is not needed for subscription tracking, such as host names and IP addresses.
endif::[]

Disconnected users::
If you cannot or do not want to connect your system directly, you can export a local summary of your product usage and import it into a CMDB, spreadsheet, or another tracking tool.

Disconnected ({Project})::
Export usage data in one of the following ways:

** Through the `foreman_rh_cloud` plugin, which generates its report locally and allows you to download it from the {ProjectWebUI}.  
This report is in JSON format and is most suitable for automated scripts or machine ingestion. 
You can generate this report by using CLI:
+
[options="nowrap" subs="+quotes,verbatim,attributes"]
----
# foreman-rake rh_cloud_inventory:report:generate
----

** Through the *Host - Installed Products* report by navigating to *Monitor* > *Reports* > *Report Templates*. 

> This report can be exported in multiple formats, including CSV, HTML, and YAML, for use in spreadsheet tools or other applications.

Reviewer 4: Is there steps to upload it to console.redhat.com for disconnected customers?

Author: Will use the link we agreed to use on Slack.

.Future-dated subscriptions
You can use future-dated subscriptions in a subscription manifest.
When you add future-dated subscriptions to your manifest before the expiry date of the existing subscriptions, you can have uninterrupted access to repositories.

.Subscriptions service capabilities
The Subscriptions service is a service on the {RHCloud} that helps organizations track and manage their Red{nbsp}Hat subscription usage across connected and disconnected environments. 

* Allows customers to view subscription usage.
* When configured, it displays usage data from both connected and disconnected environments.
* Tracks Red{nbsp}Hat product usage and capacity at the organization or account level.
* Generates a daily snapshot of usage data across all tracked systems.
* Displays data in a filterable dashboard at the {RHCloud}.
* Collects historical data for analytics, purchasing decisions, capacity planning, and risk mitigation.
* Supports the transition to Simple Content Access (SCA), which enforces subscriptions across the organization rather than on individual systems.

> .Configuring your system to use the Subscriptions service

Reviewer 2: I think this should be split off to a procedure module.

ifndef::satellite[]
.Prerequisites
* Enable the `foreman_rh_cloud` plugin:
[subs="nowrap" subs="+quotes,verbatim,attributes"]
----
# foreman-installer --enable-foreman-plugin-rh-cloud
----

> Connected {Project} Users::

Reviewer 2: ```suggestion
Connected {Project}::
```

Configure the `foreman_rh_cloud` plugin to share inventory information with the Hybrid Cloud Console.
Ensure that the *Automatic Inventory Upload* setting is enabled.
The plugin enables the subscriptions service to track usage information across connected systems.

You can configure the plugin to omit data that is not needed for subscription tracking, such as host names and IP addresses.

> endif::[]

Reviewer 2: The `endif` is in a wrong place.

Disconnected users::

> If you cannot or do not want to connect your system directly, you can export a local summary of your product usage and import it into a CMDB, spreadsheet, or another tracking tool.

Reviewer 2: I don't understand what this has to do with anything in this documentation. Do we have to keep it?

Author: I don't think it's vital to the documentation here. This is more marketing level content here. I'll remove it.

> Disconnected ({Project})::

Reviewer 2: ```suggestion
Disconnected {Project}::
```
